### PR TITLE
chore(dependencies): Preliminary work to integrate `updatecli`.

### PR DESCRIPTION
**Pull request by gounthar**
After reading @MarkEWaite's comment on PR #1707, I've decided to prepare the addition of updatecli to monitor Java versions. As part of this, I've included a few ARG instructions in the Dockerfiles to set the Java version.

I created another PR (#1711) with `updatecli` manifests.

### Testing done

```bash
make build-alpine_jdk21
make build-debian_jdk21
make build-debian_slim_jdk21
make build-rhel_ubi9_jdk21
make build-alpine_jdk17
make build-debian_jdk17
make build-debian_slim_jdk17
make build-rhel_ubi9_jdk17
make build-alpine_jdk11
make build-debian_jdk11
make build-debian_slim_jdk11
make build-rhel_ubi8_jdk11
```

### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- ~~Link to relevant issues in GitHub or Jira~~
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by dduportal** [comment]
Quick note: As discussed together, we can eventually keep the `FROM` if the version is only used here (vs. using an `ARG` with the version if the version is needed twice or more in the Dockerfile). I did it in https://github.com/dduportal/jenkins-examples/pull/65/files and I believe it can help to simplify some updatecli manifest by factorizing the `transformers` to the source itself (as it would be also reused for the `condition` step for checking if the Docker image exists).

**Comment by hervelemeur** [comment]
Should the corresponding parts in dependabot config be removed too?

File: 11/centos/centos7/hotspot/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,5 @@
-FROM eclipse-temurin:11.0.20.1_1-jdk-centos7 as jre-build
+ARG JAVA_VERSION=11.0.20.1_1
+FROM eclipse-temurin:"$JAVA_VERSION"-jdk-centos7 as jre-build
```
Comment (dduportal) [inline 11/centos/centos7/hotspot/Dockerfile:2]:
```suggestion
FROM eclipse-temurin:"${JAVA_VERSION}"-jdk-centos7 as jre-build
```

To stay coherent with the Dockerfile already using an `ARG` (and it is a good practise)

File: 17/alpine/hotspot/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
-ARG JAVA_VERSION="17.0.8.1_1"
 ARG ALPINE_TAG=3.18.3
-FROM eclipse-temurin:"${JAVA_VERSION}"-jdk-alpine AS jre-build
+ARG JAVA_VERSION=17.0.8.1_1
+FROM eclipse-temurin:"$JAVA_VERSION"-jdk-alpine AS jre-build
```
Comment (dduportal) [inline 17/alpine/hotspot/Dockerfile:3]:
```suggestion
FROM eclipse-temurin:"${JAVA_VERSION}"-jdk-alpine AS jre-build
```

To stay coherent with the Dockerfile already using an `ARG` (and it is a good practise)

File: 17/debian/bookworm-slim/hotspot/Dockerfile [hunk 3]
```diff
@@ -1,6 +1,6 @@
-ARG JAVA_VERSION="17.0.8.1_1"
 ARG BOOKWORM_TAG=20230904
-FROM eclipse-temurin:"${JAVA_VERSION}"-jdk-jammy as jre-build
+ARG JAVA_VERSION=17.0.8.1_1
+FROM eclipse-temurin:"$JAVA_VERSION"-jdk-jammy as jre-build
```
Comment (dduportal) [inline 17/debian/bookworm-slim/hotspot/Dockerfile:3]:
```suggestion
FROM eclipse-temurin:"${JAVA_VERSION}"-jdk-jammy as jre-build
```

To stay coherent with the Dockerfile already using an `ARG` (and it is a good practise)

File: 17/debian/bookworm/hotspot/Dockerfile [hunk 4]
```diff
@@ -1,6 +1,6 @@
-ARG JAVA_VERSION="17.0.8.1_1"
 ARG BOOKWORM_TAG=20230904
-FROM eclipse-temurin:"${JAVA_VERSION}"-jdk-jammy as jre-build
+ARG JAVA_VERSION=17.0.8.1_1
+FROM eclipse-temurin:"$JAVA_VERSION"-jdk-jammy as jre-build
```
Comment (dduportal) [inline 17/debian/bookworm/hotspot/Dockerfile:3]:
```suggestion
FROM eclipse-temurin:"${JAVA_VERSION}"-jdk-jammy as jre-build
```

File: 21/alpine/hotspot/Dockerfile [hunk 5]
```diff
@@ -6,6 +6,7 @@ ARG TARGETPLATFORM
 
 # hadolint ignore=DL4006
 RUN apk add --no-cache jq wget \
+  && BUILD_NUMBER=$(echo $JAVA_VERSION | cut -d'+' -f2) \
```
Comment (dduportal) [inline 21/alpine/hotspot/Dockerfile:9]: Damn, I messed up my comments and I accidentally commented in https://github.com/jenkinsci/docker/pull/1711/files#r1335046154.

Do you mind cherry-picking here?

File: 21/debian/bookworm-slim/hotspot/Dockerfile [hunk 6]
```diff
@@ -11,6 +11,7 @@ RUN set -x; apt-get update \
     ca-certificates \
     jq \
     wget \
+  && BUILD_NUMBER=$(echo $JAVA_VERSION | cut -d'+' -f2) \
```
Comment (dduportal) [inline 21/debian/bookworm-slim/hotspot/Dockerfile:14]: Damn, I messed up my comments and I accidentally commented in https://github.com/jenkinsci/docker/pull/1711/files#r1335046154.

Do you mind cherry-picking here?

File: 21/debian/bookworm/hotspot/Dockerfile [hunk 7]
```diff
@@ -11,6 +11,7 @@ RUN set -x; apt-get update \
     ca-certificates \
     jq \
     wget \
+  && BUILD_NUMBER=$(echo $JAVA_VERSION | cut -d'+' -f2) \
```
Comment (dduportal) [inline 21/debian/bookworm/hotspot/Dockerfile:14]: Damn, I messed up my comments and I accidentally commented in https://github.com/jenkinsci/docker/pull/1711/files#r1335046154.

Do you mind cherry-picking here?

File: 21/rhel/ubi9/hotspot/Dockerfile [hunk 8]
```diff
@@ -6,6 +6,7 @@ ARG TARGETPLATFORM
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 RUN set -x; yum install -y jq wget \
+  && BUILD_NUMBER=$(echo $JAVA_VERSION | cut -d'+' -f2) \
```
Comment (dduportal) [inline 21/rhel/ubi9/hotspot/Dockerfile:9]: Damn, I messed up my comments and I accidentally commented in https://github.com/jenkinsci/docker/pull/1711/files#r1335046154.

Do you mind cherry-picking here?

File: 21/alpine/hotspot/Dockerfile (hunk 5 above)
Comment (gounthar) [reply 21/alpine/hotspot/Dockerfile:9]: No problem. 🤗

**Review by dduportal — APPROVED** [review]
LGTM!

**Pull request merged**